File: easy/FibonacciNumber.js
// The Fibonacci numbers, commonly denoted F(n) form a sequence, called the Fibonacci sequence, such that each number is the sum of the two preceding ones, starting from 0 and 1. That is,

var fib = function(n) {
  if (n < 2) {
    return n;
  }
  let a = 0;
  let b = 1;
  for (let i = 1; i < n; i++) {
    let temp = b;
    b = b + a;
    a = temp;
  }
  return b;
};


console.log(fib(0))
console.log(fib(1))
console.log(fib(2))
console.log(fib(3))
console.log(fib(4))
console.log(fib(5))
console.log(fib(6))
console.log(fib(7))
console.log(fib(8))
console.log(fib(9))
console.log(fib(10))
console.log(fib(11))

File: easy/FibonacciNumber.rb
# The Fibonacci numbers, commonly denoted F(n) form a sequence, called the Fibonacci sequence, such that each number is the sum of the two preceding ones, starting from 0 and 1. That is,

# F(0) = 0, F(1) = 1
# F(n) = F(n - 1) + F(n - 2), for n > 1.
# Given n, calculate F(n).

def fib(n)
  if n < 2
      return n
  end
  a = 0
  b = 1
  for i in 1...n
      temp = a + b
      a = b
      b = temp
  end
  return b
end

puts fib(3) == 2
puts fib(10) == 55